include/GIL/Module.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef GLU_GIL_MODULE_HPP_
+#define GLU_GIL_MODULE_HPP_
+
+#include <list>
+#include <string>
+#include <memory>
+
+
+namespace glu::gil {
+
+class Function;
+
+class Type;
+
+/// @brief This class represents modules in our gil intermediate representation
+/// Modules are used to automaticaly import all delcarations in a file,
+/// see https://glu-lang.org/modules/
+class Module {
+public:
+    /// @brief Constructor for a Module instance that initialise its name
+    /// @param moduleName A string representing the modules name
+    Module(std::string moduleName);
+
+    /// @brief Adds a Function to the Functions dedicated chained list
+    /// @param function A pointer to the function that will be added
+    void addFunction(Function *function);
+
+    /// @brief Adds a Module to the Modules dedicated chained list
+    /// @param module A pointer to the module that will be added
+    void addModule(Module *module);
```

Comment:
why add a module to a module
explain
(also, imports are planned for October 2025, so next year)

Reply:
okay I'll add an explanation in the oxygen.